tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/ChangelogUpdateTool.cs

```diff
@@ -0,0 +1,197 @@
+
+                if (isManagementPlane)
+                {
+                    return await HandleManagementPlaneChangelogAsync(sdkRepoRoot, packagePath, ct);
+                }
+                else
+                {
+                    return HandleDataPlaneChangelog();
+                }
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error occurred while updating changelog for package at: {PackagePath}", packagePath);
+                return CreateFailureResponse<DefaultCommandResponse>($"An error occurred: {ex.Message}");
+            }
+        }
+
+        /// <summary>
+        /// Handles changelog update for management-plane packages by invoking the update-changelog script.
+        /// </summary>
+        /// <param name="sdkRepoRoot">Root path of the SDK repository.</param>
+        /// <param name="packagePath">Path to the package directory.</param>
+        /// <param name="ct">Cancellation token for the operation.</param>
+        /// <returns>A response indicating the result of the script execution.</returns>
+        private async Task<DefaultCommandResponse> HandleManagementPlaneChangelogAsync(
+            string sdkRepoRoot,
+            string packagePath,
+            CancellationToken ct)
+        {
+            // Prepare script parameters
+            var scriptParameters = new Dictionary<string, string>
+            {
+                { "SdkRepoPath", sdkRepoRoot },
+                { "PackagePath", packagePath }
+            };
+
+            // Create and execute process options for the update-changelog script
+            var processOptions = await CreateProcessOptions(ConfigType.UpdateChangelog, sdkRepoRoot, packagePath, scriptParameters, 15);
+            return await ExecuteProcessAsync(processOptions, ct, "Changelog content is updated.", ["Update the version if it's a release."]);
+        }
+
+        /// <summary>
+        /// Handles changelog guidance for data-plane packages.
+        /// </summary>
+        /// <returns>A noop response with guidance for manual editing.</returns>
+        private DefaultCommandResponse HandleDataPlaneChangelog()
+        {
+            logger.LogInformation("Data-plane package detected, providing manual editing guidance");
+            
+            return new DefaultCommandResponse
+            {
+                Result = "noop",
+                Message = "Data-plane changelog untouched; manual edits required.",
+                NextSteps = ["Update the version if it's a release."]
+            };
+        }
+
+        /// <summary>
+        /// Determines the SDK repository root from the package path.
+        /// </summary>
+        /// <param name="packagePath">The package path to start searching from.</param>
+        /// <returns>The SDK repository root path, or null if not found.</returns>
+        private string? DetermineSdkRepositoryRoot(string packagePath)
```

Comment:
Can you move this to PackageHelper. I think this will be helpful for other tools.

Reply:
We already have a helper for this in GitHelper

Reply:
Any reason not to use DiscoverRepoRoot in GitHelper?
https://github.com/Azure/azure-sdk-tools/blob/main/tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/GitHelper.cs#L177C21-L177C38?